.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,53 @@
+name: End to End integration tests
+
+on:
+  - pull_request
+  - push
+  - workflow_dispatch
+
+jobs:
+  integration-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout rucio containers repository
+        uses: actions/checkout@v2
+        with:
+          repository: rucio/containers
+      - uses: actions/checkout@v2
+        name: Checkout rucio source
+        with:
+          path: dev/rucio
+      - name: Prepare Images
+        shell: bash
+        run: |
+          sed -i "s/RUN git clone .*/COPY .\/rucio \/tmp\/rucio/" $GITHUB_WORKSPACE/dev/Dockerfile_py3
+      - name: Build Images
+        uses: whoan/docker-build-with-cache-action@v5
```

Comment:
as a general question: is it fine to trust third party code with the GITHUB_TOKEN ? Malicious code could edit issues/PRs/source code on the repo. So far I only trusted the GitHub curated actions. CC: @bari12

Reply:
I'm not particularly comfortable with this. Is there a safer way to do this?

Reply:
Sure, I've removed it